astro/tests/integration_test.go

```diff
@@ -52,6 +60,75 @@ func getSessionDirs(sessionBaseDir string) ([]string, error) {
 	return sessionDirs, nil
 }
 
+// compiles the astro binary and returns the path to it.
+func compileAstro(dir string) (string, error) {
+	astroPath := filepath.Join(dir, "astro")
+	packageName := "github.com/uber/astro/astro/cli/astro"
+	out, err := exec.Command("go", "build", "-o", astroPath, packageName).CombinedOutput()
+	if err != nil {
+		return "", errors.New(string(out))
+	}
+	return astroPath, nil
+}
+
+func TestPlanInterrupted(t *testing.T) {
+	fakeTerraformPath := "fixtures/terraform"
+	require.True(t, utils.FileExists(fakeTerraformPath))
+	fakeTerraformDir, err := filepath.Abs(filepath.Dir(fakeTerraformPath))
+	require.NoError(t, err)
+
+	oldPath := os.Getenv("PATH")
+	os.Setenv("PATH", fmt.Sprintf("%s:%s", fakeTerraformDir, oldPath))
+	defer os.Setenv("PATH", oldPath)
+
+	tmpdir, err := ioutil.TempDir("", "astro-binary")
+	defer os.RemoveAll(tmpdir)
+	require.NoError(t, err)
+
+	astroBinary, err := compileAstro(tmpdir)
+	require.NoError(t, err)
+	command := exec.Command(astroBinary, "plan")
+
+	fixtureAbsPath, err := filepath.Abs("fixtures/plan-interrupted")
+	require.NoError(t, err)
+	command.Dir = fixtureAbsPath
+
+	stdoutBytes := &bytes.Buffer{}
+	stderrBytes := &bytes.Buffer{}
+	command.Stdout = stdoutBytes
+	command.Stderr = stderrBytes
+
+	var cmdErr error
+	processChan := make(chan struct{}, 1)
+	go func() {
+		defer close(processChan)
+		cmdErr = command.Run()
+		processChan <- struct{}{}
+	}()
+
+	// let astro start terraform processes
+	time.Sleep(1000 * time.Millisecond)
```

Comment:
How was this threshold chosen?

Reply:
This was just enough to start the astro process and I didn't want this test to be flaky if we decrease the number.